test/DataRequirementHelpers.test.ts

```diff
@@ -511,7 +511,68 @@ describe('DataRequirementHelpers', () => {
         ],
         dateFilter: [
           {
-            path: 'performed.end',
+            path: 'authoredOn.end',
+            valuePeriod: {
+              start: '2019-01-01',
+              end: '2019-12-31'
+            }
+          },
+          {
+            path: 'occurrence.start',
+            valuePeriod: {
+              start: '2019-01-01',
+              end: '2019-12-31'
+            }
+          }
+        ]
+      };
+      DataRequirementHelpers.addFhirQueryPatternToDataRequirements(testDataReqWithDateFilter);
+      expect(testDataReqWithDateFilter.extension?.length).toEqual(2);
+      if (testDataReqWithDateFilter.extension) {
+        expect(testDataReqWithDateFilter.extension[0].valueString).toEqual(
+          '/ServiceRequest?code:in=http://example.com&status=completed&authored=ge2019-01-01&authored=le2019-12-31&occurrence=ge2019-01-01&occurrence=le2019-12-31&subject=Patient/{{context.patientId}}'
+        );
+        expect(testDataReqWithDateFilter.extension[1].valueString).toEqual(
+          '/ServiceRequest?code:in=http://example.com&status=completed&authored=ge2019-01-01&authored=le2019-12-31&occurrence=ge2019-01-01&occurrence=le2019-12-31&performer=Patient/{{context.patientId}}'
+        );
+      }
+    });
+
+    test('add fhirQueryPattern extension with dateTime and duration date filters', () => {
+      const testDataReqWithDateFilter: DataRequirement = {
+        type: 'ServiceRequest',
+        dateFilter: [
+          {
+            path: 'authoredOn',
+            valueDateTime: '2019-01-01'
+          },
+          {
+            path: 'occurrence', // Note: duration is probably not a practical measure of occurrence here
+            valueDuration: {
+              unit: 'days',
+              value: 10
+            }
+          }
+        ]
+      };
+      DataRequirementHelpers.addFhirQueryPatternToDataRequirements(testDataReqWithDateFilter);
+      expect(testDataReqWithDateFilter.extension?.length).toEqual(2);
+      if (testDataReqWithDateFilter.extension) {
+        expect(testDataReqWithDateFilter.extension[0].valueString).toEqual(
```

Comment:
Nitpicking, but I think it's better to just continue to use `?.` chaining here rather than stick these inside an if

Reply:
Great call